src/main/java/com/zufar/onlinestore/security/signin/attempts/LoginAttemptManager.java

```diff
@@ -0,0 +1,59 @@
+package com.zufar.onlinestore.security.signin.attempts;
+
+import com.zufar.onlinestore.security.signin.attempts.entity.LoginAttemptEntity;
+import com.zufar.onlinestore.security.exception.AccountLockedException;
+import com.zufar.onlinestore.security.signin.attempts.repository.LoginAttemptRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDateTime;
+import java.util.Optional;
+
+@Service
+@RequiredArgsConstructor
+public class LoginAttemptManager {
+
+    private static final int MAX_LOGIN_ATTEMPTS = 5;
+    private static final int LOCKOUT_DURATION_MINUTES = 30;
+    private static final int INITIAL_LOGIN_ATTEMPTS_COUNT = 1;
+
+    private final LoginAttemptRepository loginAttemptRepository;
+
+    public void handleFailedLogin(final String userEmail) {
+        Optional<LoginAttemptEntity> existingAttempt = loginAttemptRepository.findByUserEmail(userEmail);
+
+        LoginAttemptEntity loginAttempt;
+        if (existingAttempt.isEmpty()) {
```

Comment:
pls optimize this code using new java 9+ optional api to get rid of non-relevant if-elses blocks

Reply:
Fixed